src/utils/hooks/useGetFullApplicationStructure.tsx

```diff
@@ -49,6 +50,13 @@ const useGetFullApplicationStructure = ({
     fetchPolicy: networkFetch ? 'network-only' : 'cache-first',
   })
 
+  // TODO - might need a use effect if serial is changes (navigated to another application from current page)
+
+  useEffect(() => {
+    if (!data) return
+    setLastRefetchedTimestamp(Date.now())
+  }, [data])
+
   useEffect(() => {
     if (loading) {
```

Comment:
I don't think this one is needed anymore

Reply:
so right 

c646ba662da02bf59768d000de2d11ce7f7f8b66